src/tangd.c

```diff
@@ -184,19 +188,39 @@ rec(http_method_t method, const char *path, const char *body,
                       "\r\n%s", strlen(enc), enc);
 }
 
-static struct http_dispatch dispatch[] = {
-    { adv, 1 << HTTP_GET,  2, "^/+adv/+([0-9A-Za-z_-]+)$" },
-    { adv, 1 << HTTP_GET,  2, "^/+adv/*$" },
-    { rec, 1 << HTTP_POST, 2, "^/+rec/+([0-9A-Za-z_-]+)$" },
-    {}
-};
-
 #define DEFAULT_PORT 9090
 
+static struct http_dispatch s_dispatch[] = {
+        { adv, 1 << HTTP_GET,  2, NULL},
```

Comment:
Instead of `NULL`, why don't you leave as it was here, i.e., the default (with no `endpoint`), and then you adjust it in the case there is an endpoint specified?